### ISSUE TITLE
Empty lines are not preserved 

### ISSUE THREAD

**Issue by feckert** [jirutka/esh#10]
First of all great job
I have always been looking for a template engine that works only with shell binaries.
The advantage for me is I don't need to install any annoying dependencies like with other template engines for example on python.

But I have a problem. The new lines are not present on the output if the preview line is a esh tag.
I am working with openwrt and I do have a lot off image. So that I do not have duplicated configuration files.
I sqashed them in one file as an template for esh to generate the different uci image configuration out of this file.
It works as expected, with a small bug I think.

The new line ```--><---  NEWLINE MISSING IN OUTPUT``` (see: Example code) is not present on the output. And if i remove the `-` in the ```if``` statement the line above ```<% fi %>```. I get two new lines! So I do not know what is the problem. Is this maybe a bug?

Example code:
```
config interface 'lan'
<% if [ "$common_customer" = "abc" ]; then -%>
	option proto 'none'
<% else -%>
	option proto 'static'
	option netmask '255.255.255.0'
<% fi -%>
	option ifname 'eth1.1'
	option force_link '1'
<% if [ "$common_customer" = "123" ]; then -%>
	option ipaddr '192.168.0.1'
<% elif [ "$common_customer" = "xyc" ]; then -%>
	option ipaddr '192.168.0.254'
<% elif [ "$common_customer" = "abc" ]; then -%>
<% else -%>
	option ipaddr '192.168.0.50'
<% fi -%>
--><--- NEWLINE MISSING IN OUTPUT
config interface 'wan'
	option ifname 'eth0'
<% if [ "$network_wan_auto" != "yes" ]; then -%>
	option auto '0'
<% fi -%>
<% if [ "$network_wan_metric" != "default" ]; then -%>
	option metric '<%= $network_wan_metric %>'
<% else -%>
<% if [ "$common_customer" != "123" ]; then -%>
	option metric '100'
<% fi -%>
<% fi -%>
	option proto 'dhcp'
```

**Comment by jirutka** [comment]
Thanks for kind words!

I’ve fixed it in  6899cc6. Please try it on your templates and let me know if everything is right.

**Comment by feckert** [comment]
Thanks for the quick fix. I seems to work as expected.
Thanks :+1:
